kythe/go/serving/xrefs/xrefs.go

```diff
@@ -624,6 +627,53 @@ func decorationToReference(norm *span.Normalizer, d *srvpb.FileDecorations_Decor
 	}
 }
 
+type xrefCategory int
+
+const (
+	xrefCategoryNone xrefCategory = iota
+	xrefCategoryDef
+	xrefCategoryDecl
+	xrefCategoryRef
+	xrefCategoryCall
+	xrefCategoryRelated
+	xrefCategoryIndirection
+)
+
+func (c xrefCategory) AddCount(reply *xpb.CrossReferencesReply, idx *srvpb.PagedCrossReferences_PageIndex, pageSet *pageSet) {
+	switch c {
+	case xrefCategoryDef:
+		if pageSet.Contains(idx) {
+			reply.Total.Definitions += int64(idx.Count)
+		} else {
+			reply.Filtered.Definitions += int64(idx.Count)
+		}
+	case xrefCategoryDecl:
+		if pageSet.Contains(idx) {
+			reply.Total.Declarations += int64(idx.Count)
+		} else {
+			reply.Filtered.Declarations += int64(idx.Count)
+		}
+	case xrefCategoryRef:
+		if pageSet.Contains(idx) {
+			reply.Total.References += int64(idx.Count)
+		} else {
+			reply.Filtered.References += int64(idx.Count)
+		}
```

Comment:
should this be?
```
case xrefCategoryDef, xrefCategoryDecl, xrefCategoryRef:
		if pageSet.Contains(idx) {
			reply.Total.Definitions += int64(idx.Count)
		} else {
			reply.Filtered.Definitions += int64(idx.Count)
		}
```

Reply:
I don't understand.  Why would decls and refs both be counted as defs?